doc/api/esm.md

```diff
@@ -177,107 +197,114 @@ import 'data:text/javascript,console.log("hello!");';
+```js
+import fs, { readFileSync } from 'fs';
+import { syncBuiltinESMExports } from 'module';
 
-`NODE_PATH` is not part of resolving `import` specifiers. Please use symlinks
-if this behavior is desired.
+fs.readFileSync = () => Buffer.from('Hello, ESM');
+syncBuiltinESMExports();
 
-### No `require`, `exports`, `module.exports`, `__filename`, `__dirname`
+fs.readFileSync === readFileSync;
+```
 
-These CommonJS variables are not available in ES modules.
+## `import.meta`
 
-`require` can be imported into an ES module using [`module.createRequire()`][].
+* {Object}
 
-Equivalents of `__filename` and `__dirname` can be created inside of each file
-via [`import.meta.url`][].
+The `import.meta` meta property is an `Object` that contains the following
+properties.
 
-```js
-import { fileURLToPath } from 'url';
-import { dirname } from 'path';
+### `import.meta.url`
 
-const __filename = fileURLToPath(import.meta.url);
-const __dirname = dirname(__filename);
+* {string} The absolute `file:` URL of the module.
+
+This is defined exactly the same as it is in browsers providing the URL of the
+current module file.
+
+This enables useful patterns such as relative file loading:
+
+```js
+import { readFileSync } from 'fs';
+const buffer = readFileSync(new URL('./data.proto', import.meta.url));
 ```
 
-### No `require.resolve`
+### `import.meta.resolve(specifier[, parent])`
 
-Former use cases relying on `require.resolve` to determine the resolved path
-of a module can be supported via `import.meta.resolve`, which is experimental
-and supported via the `--experimental-import-meta-resolve` flag:
+> Stability: 1 - Experimental
 
+* `specifier` {string} The module specifier to resolve relative to `parent`.
+* `parent` {string|URL} The absolute parent module URL to resolve from. If none
+  is specified, the value of `import.meta.url` is used as the default.
+* Returns: {Promise}
+
+<!-- eslint-skip -->
 ```js
-(async () => {
-  const dependencyAsset = await import.meta.resolve('component-lib/asset.css');
-})();
+const dependencyAsset = await import.meta.resolve('component-lib/asset.css');
 ```
 
 `import.meta.resolve` also accepts a second argument which is the parent module
 from which to resolve from:
 
+<!-- eslint-skip -->
 ```js
-(async () => {
-  // Equivalent to import.meta.resolve('./dep')
-  await import.meta.resolve('./dep', import.meta.url);
-})();
+await import.meta.resolve('./dep', import.meta.url);
 ```
 
 This function is asynchronous because the ES module resolver in Node.js is
-asynchronous. With the introduction of [Top-Level Await][], these use cases
-will be easier as they won't require an async function wrapper.
+asynchronous.
 
-### No `require.extensions`
+## `import()` expressions
 
-`require.extensions` is not used by `import`. The expectation is that loader
-hooks can provide this workflow in the future.
-
-### No `require.cache`
-
-`require.cache` is not used by `import`. It has a separate cache.
-
-### URL-based paths
```

Comment:
This is the only heading that was removed, which is consolidated into the new `file:` URL section.